File: be1-go/message/query/method/greet_server.go
package method

import "popstellar/message/query"

type ServerInfo struct {
	PublicKey     string `json:"public_key"`
	ServerAddress string `json:"server_address"`
	ClientAddress string `json:"client_address"`
}

// GreetServer defines a JSON RPC greetServer message
type GreetServer struct {
	query.Base

	ID int `json:"id"`

	Params ServerInfo `json:"params"`
}

File: be2-scala/src/main/scala/ch/epfl/pop/model/network/method/GreetServer.scala
package ch.epfl.pop.model.network.method
import ch.epfl.pop.json.HighLevelProtocol._
import ch.epfl.pop.model.network.Parsable
import ch.epfl.pop.model.objects.{Channel, Hash, PublicKey}
import spray.json._
final case class GreetServer(publicKey: PublicKey, clientAddress: String, serverAddress: String) extends Params {
  override def hasChannel: Boolean = false

  override def hasMessage: Boolean = false
}

object GreetServer extends Parsable {

  def apply(publicKey: PublicKey, clientAddress: String, serverAddress: String): GreetServer = {
    new  GreetServer(publicKey, clientAddress, serverAddress)
  }

  override def buildFromJson(payload: String): Any = payload.parseJson.asJsObject.convertTo[GreetServer]
}
